lib/pairmotron/pair_maker.ex

```diff
@@ -58,23 +62,23 @@ defmodule Pairmotron.PairMaker do
       |> Mixer.mixify(week)
       |> Pairer.generate_pairs(pairs)
 
-    results.pairs
-      |> Enum.map(fn(users) -> make_pairs(users, year, week, group_id) end)
-      |> List.flatten
-      |> Enum.map(fn(p) -> Repo.insert! p end)
+    multi = results.pairs
+      |> Enum.reduce(multi, fn(users, acc) ->
+          insert_pair(acc, year, week, group_id, users)
+      end)
 
-    insert_user_pair(results.user_pair)
+    multi = insert_lone_user_pair(multi, results.user_pair)
+    Repo.transaction(multi)
```

Comment:
I'm not sure if this should care about the success for failure of the transaction. The controller might, but I'm not sure if anything would be useful save a "Something failed during repairifying" message. Thoughts?